arelle/plugin/validate/DBA/rules/__init__.py

```diff
@@ -52,6 +52,27 @@ def errorOnDateFactComparison(
         )
 
 
+def errorOnRequiredFact(
+        modelXbrl: ModelXbrl,
+        factQn: QName,
+        code: str,
+        message: str,
+) -> Iterable[Validation]:
+    """
+    Yields an error if a fact with the given QName is not tagged with a non-nil value.
+    :return: Yields validation errors.
+    """
+    facts: set[ModelFact] = modelXbrl.factsByQname.get(factQn, set())
+    if facts:
+        for fact in facts:
+            if fact.xValue is not None:
```

Comment:
I _think_ checking for the `xsi:nil` attribute (ModelFact.isNil [here](https://github.com/Arelle/Arelle/blob/672eed35cb54b8142f2d6facdd2127ff056c0592/arelle/ModelInstanceObject.py#L353)) is what we should interpret "non-nil" to mean in the rule text. That's at least what we've done [elsewhere](https://github.com/Arelle/Arelle/blob/672eed35cb54b8142f2d6facdd2127ff056c0592/arelle/plugin/validate/DBA/rules/fr.py#L224).